ndb/tests/system/test_system.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+from google.cloud import datastore
+from google.cloud import ndb
+
+
+@pytest.fixture
+def ds_entity():
+    keys = []
+    client = datastore.Client()
+
+    def make_entity(*key_args, **entity_kwargs):
+        key = client.key(*key_args)
+        assert client.get(key) is None
+        entity = datastore.Entity(key=key)
+        entity.update(entity_kwargs)
+        client.put(entity)
+
+        keys.append(key)
+        return entity
+
+    yield make_entity
+
+    for key in keys:
+        client.delete(key)
+
+
+def test_retrieve_entity(ds_entity):
+    ds_entity("SomeKind", 1234, foo=42, bar="none")
```

Comment:
This is flake-prone, i.e. if two jobs run this system test at once, they'll collide. Instead of `1234`, use a UUID or use `test_utils.system.unique_resource_id()`.